src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java

```diff
@@ -353,56 +351,38 @@ private OutputAlleleSubset(final List<Allele> alleles, final List<Integer> mleCo
      * @return information about the alternative allele subsetting {@code null}.
      */
     private OutputAlleleSubset calculateOutputAlleleSubset(final AFCalculationResult afCalculationResult, final VariantContext vc) {
```

Comment:
As discussed in person, we fully support the mission of making this class less stateful (and particularly making it so that `isVcCoveredByDeletion()` does not modify state, which was the original ticket), but we're concerned that this method `calculateOutputAlleleSubset()` was intended to be stateful, and to accumulate deletions across calls. It's also not well covered by tests, so this change could easily break the method.

Can you think about this issue and reply here once you've looked into it, @davidbenjamin ? Also at-mentioning @ronlevine to solicit his comments, since he appears to have added `upstreamDeletionsLoc` to this class in the first place.